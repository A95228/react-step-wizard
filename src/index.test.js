import React from 'react';
import Renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import StepWizard from './index';

Enzyme.configure({ adapter: new Adapter() });

/** Helper Functions */
const getState = (component) => {
    const wrapper = shallow(component);
    wrapper.instance();

    return wrapper.state();
};
const render = jsx => Renderer.create(jsx);
const takeSnapshot = test => expect(test).toMatchSnapshot();
const getTreeSnapshot = component => takeSnapshot(render(component));
// const getStateSnapshot = component => takeSnapshot(getState(component));
const testComponent = (component) => {
    getTreeSnapshot(component);

    const state = getState(component);
    takeSnapshot(state);

    return state;
};

/** Capture console errors */
console.error = jest.fn();

/** Unit Tests */
describe('Step Wizard Component', () => {
    it('empty component', () => {
        testComponent(<StepWizard />);
    });

    it('with 3 basic components', () => {
        testComponent((
            <StepWizard>
                <Step1 />
                <Step2 />
                <Step3 />
            </StepWizard>
        ));
    });

    it('initialStep set to 2', () => {
        testComponent((
            <StepWizard initialStep={2}>
                <Step1 />
                <Step2 />
                <Step3 />
            </StepWizard>
        ));
    });

    it('isHashEnabled - true', () => {
        testComponent((
            <StepWizard isHashEnabled>
                <Step1 hashKey={'First'} />
                <Step2 />
                <Step3 hashKey={'The End!'} />
            </StepWizard>
        ));
    });

    it('isHashEnabled - false', () => {
        testComponent((
            <StepWizard>
                <Step1 hashKey={'First'} />
                <Step2 />
                <Step3 hashKey={'The End!'} />
            </StepWizard>
        ));
    });

    it('garbage props', () => {
        console.error.mockClear();

        testComponent((
            <StepWizard
                initialStep={true}
                isHashEnabled={'fdjlk'}
                onStepChange={true}
                transitions={false}
            >
                <Step1 hashKey={'First *3@ #$~!,";'} />
                <Step2 />
                <Step3 hashKey={1} />
            </StepWizard>
        ));

        expect(console.error).toHaveBeenCalledWith('Warning: Failed prop type: Invalid prop `initialStep` of type `boolean` supplied to `StepWizard`, expected `number`.\n    in StepWizard');
        expect(console.error).toHaveBeenCalledWith('Warning: Failed prop type: Invalid prop `isHashEnabled` of type `string` supplied to `StepWizard`, expected `boolean`.\n    in StepWizard');
        expect(console.error).toHaveBeenCalledWith('Warning: Failed prop type: Invalid prop `onStepChange` of type `boolean` supplied to `StepWizard`, expected `function`.\n    in StepWizard');
        expect(console.error).toHaveBeenCalledWith('Warning: Failed prop type: Invalid prop `transitions` of type `boolean` supplied to `StepWizard`, expected `object`.\n    in StepWizard');
    });
});

describe('Step Wizard Functions', () => {
    // it('test', async () => {
    //     const wrapper = shallow(<StepWizard />);
    //     await wrapper.instance();

    //     expect(wrapper.state()).toEqual({});
    // });
});

const Step1 = () => <div>Step 1</div>;
const Step2 = () => <div>Step 2</div>;
const Step3 = () => <div>Step 3</div>;
