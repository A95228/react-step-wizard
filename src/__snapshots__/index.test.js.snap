// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step Wizard Component empty component 1`] = `
<div
  className="step-wizard"
/>
`;

exports[`Step Wizard Component empty component 2`] = `
Object {
  "activeStep": 0,
  "classes": Object {
    "0": "",
  },
  "hashKeys": Object {},
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Component garbage props 1`] = `
<div
  className="step-wizard"
>
  <div
    className="step "
  >
    <div>
      Step 1
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 2
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 3
    </div>
  </div>
</div>
`;

exports[`Step Wizard Component garbage props 2`] = `
Object {
  "activeStep": 0,
  "classes": Object {
    "0": "",
    "1": "hide",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "First *3@ #$~!,\\";",
    "1": 2,
    "2": 1,
    "First *3@ #$~!,\\";": 0,
    "step2": 1,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Component initialStep set to 2 1`] = `
<div
  className="step-wizard"
>
  <div
    className="step hide"
  >
    <div>
      Step 1
    </div>
  </div>
  <div
    className="step "
  >
    <div>
      Step 2
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 3
    </div>
  </div>
</div>
`;

exports[`Step Wizard Component initialStep set to 2 2`] = `
Object {
  "activeStep": 1,
  "classes": Object {
    "0": "hide",
    "1": "",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "step1",
    "1": "step2",
    "2": "step3",
    "step1": 0,
    "step2": 1,
    "step3": 2,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Component isHashEnabled - false 1`] = `
<div
  className="step-wizard"
>
  <div
    className="step "
  >
    <div>
      Step 1
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 2
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 3
    </div>
  </div>
</div>
`;

exports[`Step Wizard Component isHashEnabled - false 2`] = `
Object {
  "activeStep": 0,
  "classes": Object {
    "0": "",
    "1": "hide",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "First",
    "1": "step2",
    "2": "The End!",
    "First": 0,
    "The End!": 2,
    "step2": 1,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Component isHashEnabled - true 1`] = `
<div
  className="step-wizard"
>
  <div
    className="step "
  >
    <div>
      Step 1
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 2
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 3
    </div>
  </div>
</div>
`;

exports[`Step Wizard Component isHashEnabled - true 2`] = `
Object {
  "activeStep": 0,
  "classes": Object {
    "0": "",
    "1": "hide",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "First",
    "1": "step2",
    "2": "The End!",
    "First": 0,
    "The End!": 2,
    "step2": 1,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Component isLazyMount - true 1`] = `
<div
  className="step-wizard"
>
  <div
    className="step "
  >
    <div>
      Step 2
    </div>
  </div>
</div>
`;

exports[`Step Wizard Component isLazyMount - true 2`] = `
Object {
  "activeStep": 1,
  "classes": Object {
    "0": "hide",
    "1": "",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "step1",
    "1": "step2",
    "2": "step3",
    "step1": 0,
    "step2": 1,
    "step3": 2,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Component nav 1`] = `
<div
  className="step-wizard"
>
  <div
    currentStep={1}
    firstStep={[Function]}
    goToStep={[Function]}
    isActive={false}
    lastStep={[Function]}
    nextStep={[Function]}
    previousStep={[Function]}
    totalSteps={3}
    transitions="hide"
  >
    Nav
  </div>
  <div
    className="step "
  >
    <div>
      Step 1
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 2
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 3
    </div>
  </div>
</div>
`;

exports[`Step Wizard Component nav 2`] = `
Object {
  "activeStep": 0,
  "classes": Object {
    "0": "",
    "1": "hide",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "step1",
    "1": "step2",
    "2": "step3",
    "step1": 0,
    "step2": 1,
    "step3": 2,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Component with 3 basic components 1`] = `
<div
  className="step-wizard"
>
  <div
    className="step "
  >
    <div>
      Step 1
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 2
    </div>
  </div>
  <div
    className="step hide"
  >
    <div>
      Step 3
    </div>
  </div>
</div>
`;

exports[`Step Wizard Component with 3 basic components 2`] = `
Object {
  "activeStep": 0,
  "classes": Object {
    "0": "",
    "1": "hide",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "step1",
    "1": "step2",
    "2": "step3",
    "step1": 0,
    "step2": 1,
    "step3": 2,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Functions firstStep 1`] = `
Object {
  "activeStep": 0,
  "classes": Object {
    "0": "",
    "1": "hide",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "step1",
    "1": "step2",
    "2": "step3",
    "step1": 0,
    "step2": 1,
    "step3": 2,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Functions lastStep 1`] = `
Object {
  "activeStep": 2,
  "classes": Object {
    "0": "hide animated fadeOutLeft",
    "1": "hide",
    "2": "animated fadeInRight",
  },
  "hashKeys": Object {
    "0": "step1",
    "1": "step2",
    "2": "step3",
    "step1": 0,
    "step2": 1,
    "step3": 2,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;

exports[`Step Wizard Functions nextStep 1`] = `
Object {
  "activeStep": 1,
  "classes": Object {
    "0": "hide animated fadeOutLeft",
    "1": "animated fadeInRight",
    "2": "hide",
  },
  "hashKeys": Object {
    "0": "step1",
    "1": "step2",
    "2": "step3",
    "step1": 0,
    "step2": 1,
    "step3": 2,
  },
  "transitions": Object {
    "enterLeft": "animated fadeInLeft",
    "enterRight": "animated fadeInRight",
    "exitLeft": "animated fadeOutLeft",
    "exitRight": "animated fadeOutRight",
    "intro": "",
  },
}
`;
